### DISCUSSION TITLE
Delete "esp32.NVS("label" ) or Erase complete NVS partition in ESP32

### DISCUSSION THREAD

**Discussion by nastiliano** [micropython/micropython#11163] [General]
Hello everybody:

I've been searching for a lot of pages and documentation and I cannot find the way to delete a "esp32.NVS("label" ) or Erase complete NVS partition in ESP32.
I have "boot.py" and "main.py" like this:

#BOOT.PY
param_area = esp32.NVS("param_area ")
param_area .set_blob("usuari" , "")
param_area .set_blob("cp_instal" , "")
area_param.set_i32("sen_man_mot_1" , -1)
.
.
.
param_area.commit()

#Change the name of "Boot.py" to not execute anymore
time.sleep(1)
os.rename("boot.py" , "no_boot.py")
time.sleep(2)

#MAIN.PY
.
.
.
time.sleep(1)
os.rename("no_boot.py" , "boot.py")
time.sleep(2)
machine.reset()

And an error occurs  like: "NOT ENOUGHT SPACE IN NVS".

I would need to delete the NVS space before machine.reset() not to overload my original NVS namespace.

Somebady have any ideas, please?

Thank you very much

**Comment by karfas** [comment]
See related #11023.

**Reply by nastiliano** [reply]
Hello, thanks for your response.

Yesterday I tried to do this:

" esp32.Partition.writeblocks( block_num , buf) "

But I don't know how to initialize the 2 arguments and how to erase my NVS partition namespace ( esp32.NVS("param_area ")) with this class.

Could you help me please?
Thank you

**Reply by karfas** [reply]
There is no way to erase a single "namespace", only the whole partition.
From https://docs.micropython.org/en/latest/library/esp32.html?highlight=nvs I get that `NVS.erase_key()` seems to be the correct function to remove a key and its values (maybe you get "not enough space... " due to the zero-sized data ?).

A simple way to solve your problem is (maybe) to erase either the entire flash (& install micropython and your code  again) or the NVS partition only.

**Reply by glenn20** [reply]
To overwrite the nvs partition, you could try:

```python
p = esp32.Partition.find(esp32.Partition.TYPE_DATA, label='nvs')[0]
p.writeblocks(0, bytearray(4096))
```
This will ~~erase~~ overwrite the first block in the NVS partition which I believe will cause nvs_flash_init() to fail on next reboot and micropython will automatically re-initialise the nvs partition when it boots. However - I am speculating here!!! You may need to overwrite all the blocks in the partition. It would be much cleaner if the esp32 module supported the `nvs_flash_erase()` function.

**Reply by nastiliano** [reply]
It works!!! 
SOLVED:

Thanks to everybody.

The solution to erase a NVS is like glenn20 said:

p = esp32.Partition.find(esp32.Partition.TYPE_DATA, label='nvs')[0]
p.writeblocks(0, bytearray(4096))

Unfortunately at the moment doesn't exist the nvs_flash_erase() function.
Micropython is great but sometimes I think is a little bit young yet.

Best regards!
